pkg/kv/kvserver/replica_application_decoder.go

```diff
@@ -145,7 +145,7 @@ func (d *replicaDecoder) createTracingSpans(ctx context.Context) {
 			propCtx := ctx // raft scheduler's ctx
 			var propSp *tracing.Span
 			// If the client has a trace, put a child into propCtx.
-			if sp := tracing.SpanFromContext(cmd.proposal.Context()); sp != nil {
+			if sp := tracing.SpanFromContext(cmd.proposal.Context()); sp != nil && !sp.IsNoop() {
```

Comment:
curious how you discovered this case. also lots of allocations for the line below?

Reply:
Yep, I basically went through the heap profile looking for all samples going into `tracer.StartSpanCtx`.

Reply:
Also I'll admit that this is a bit of a Rube Goldberg machine to me at this point. I'm confident that at some point I did see this line make a difference. But what were the other three things I had already turned off by the time? I'm no longer sure. There's a chance that with all of the other commits, this line doesn't make a difference. But it did at some point, similar to another check I added elsewhere (somewhere in distsql server I think) that at the time made all the difference. Ideally we accept some "spurious" fixes for now because it's quite tedious to re-validate each of them.